tests/x11/addon_products_via_SCC_yast2.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: add addon to SLES via SCC
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base qw(y2_installbase y2_module_guitest);
+use strict;
+use warnings;
+use testapi;
+use registration;
+use version_utils qw(is_sle is_leap);
+use x11utils 'turn_off_gnome_screensaver';
+use YaST::Module;
+use utils;
+
+sub test_setup {
+    select_console "x11";
+    x11_start_program('xterm');
+    turn_off_gnome_screensaver;
+}
+
+sub run {
+    my $self = shift;
+    test_setup;
+    YaST::Module::open(module => 'scc', ui => 'qt');
+    save_screenshot;
+    $testapi::distri->get_registration()->register_via_scc({
+            email => get_var('SCC_EMAIL'),
+            reg_code => get_var('SCC_REGCODE')});
+    save_screenshot;
+    my @scc_addons = split ',', get_var('SCC_ADDONS');
+    $testapi::distri->get_module_registration()->register_extension_and_modules([@scc_addons]);
+    save_screenshot;
+
+    # No libyui-rest-api for advance software selection
+    assert_screen("yast_scc-pkgtoinstall", 100);
+    wait_screen_change {
+        send_key 'alt-a';
+    };
+    assert_screen("yast_scc-installation-summary", 100);
+    save_screenshot;
+    $testapi::distri->get_module_registration_installation_report()->press_next();
+    save_screenshot;
+    assert_screen("generic-desktop", 60);
+    # Check that repos actually work
```

Comment:
could we group this in `refresh_repos` method?

Reply:
In my view no need change If you mean create sub function named refresh_repos include two commands(zypper ref & zypper repos --detail)

Reply:
`test_setup()` is the same history, they are kind of pre-requisites and post-requisites, perhaps `check_repos` is better name, but at least between line 50 and 51 one space helps to separate logical parts in the test. Same thing, trying to avoid spaghetti code hard to refactor, but this case is really minor.